### PR TITLE
feat(sprint-poker): Push task to GitLab

### LINKED ISSUE #6365: Sprint Poker: GitLab v2: Implement TaskIntegrationManager in GitLabServerManager
To be consistent with other integrations, the `GitLabServerManager` should implement the new `TaskIntegrationManager`.

See comment for more details: https://github.com/ParabolInc/parabol/pull/6267#discussion_r844481608

### AC

- [ ] `GitLabServerManager` implements the `TaskIntegrationManager`

### Updated AC:

- [ ] A user can integrate a Parabol task card with a GitLab project

### PR DESCRIPTION
**Pull request by nickoferrall**
Fix #6365 and fix https://github.com/ParabolInc/parabol/issues/6428

This PR refactors the GitLab queries and mutations to use the `TaskIntegrationManagerFactory` pattern that Igor created. Using this class, I've also implemented pushing a Parabol task card to a GitLab project. 

push-task-to-gitlab

### CODE REVIEW

File: packages/server/graphql/queries/helpers/fetchGitLabProjects.ts
Comment (nickoferrall) [inline packages/server/graphql/queries/helpers/fetchGitLabProjects.ts:19]: Sometimes we've just checked whether `provider` exists and then written `provider.serverBaseUrl!` below. Personally, I prefer to include `serverBaseUrl` in the if statement as it doesn't increase the number of lines of code and it means we don't need to use a `!`

File: packages/client/components/TaskFooterIntegrateMenu.tsx
Comment (nickoferrall) [inline packages/client/components/TaskFooterIntegrateMenu.tsx:140]: I don't love that this is so GitLab specific, but we do need the GitLab cloud provider info to be able to complete the auth [here](https://github.com/ParabolInc/parabol/blob/e11747dfe132959d3ff09221debdb2398e2b4c66/packages/client/components/AddToGitLabMenuItem.tsx#L53). 

I considered passing down `integrationsRef` instead, but as we're just using GitLab here, I felt it was best to pass down the smallest amount of data possible.

Comment (igorlesnenko) [reply packages/client/components/TaskFooterIntegrateMenu.tsx:140]: Maybe also consider passing `cloudProvider` which should be generic across different integrations

File: packages/server/integrations/TaskIntegrationManagerFactory.ts
Comment (igorlesnenko) [inline packages/server/integrations/TaskIntegrationManagerFactory.ts:72]: +1 Looks like we already checked for `auth` above

File: packages/client/components/TaskIntegrationLink.tsx
Comment (igorlesnenko) [inline packages/client/components/TaskIntegrationLink.tsx:85]: +1  Still `github` here

File: packages/client/components/TaskFooterIntegrateMenu.tsx
Comment (nickoferrall) [reply packages/client/components/TaskFooterIntegrateMenu.tsx:140]: I like it and agree that it's more generic, but `cloudProvider` can be null so `cloudProviderRef` won't work as the ref in the child fragment

File: packages/server/graphql/queries/helpers/fetchGitLabProjects.ts
Comment (Dschoordsch) [reply packages/server/graphql/queries/helpers/fetchGitLabProjects.ts:19]: You can also do `const provider = await dataLoader.get('integrationProviders').loadNonNull(providerId)` since you know it must exist if the `auth` was valid. I think the `teamMemberIntegrationAuths` loader should return the provider as well, since we always need it.

Comment (nickoferrall) [reply packages/server/graphql/queries/helpers/fetchGitLabProjects.ts:19]: `loadNonNull` guarantees that the `provider` is non-null, but the `serverBaseUrl` can still be nullable. As we have to check whether `serverBaseUrl` is null, I think there's no harm is using `if (!provider?.serverBaseUrl) return []`

File: packages/client/components/TaskFooterIntegrateMenu.tsx
Comment (nickoferrall) [inline packages/client/components/TaskFooterIntegrateMenu.tsx:43]: Check if `gitlab` exists as it doesn't yet in the demo. It'll be implemented in this issue: https://github.com/ParabolInc/parabol/issues/6443

**Comment by nickoferrall** [comment]
@mattkrick or @Dschoordsch it'd be great to get a Maintainer Review on this one when you get the chance. Once this PR and https://github.com/ParabolInc/parabol/pull/6398 are merged, we can launch v1 of the GitLab integration. 

I mentioned to @lorena-cure that I estimate that v1 will be ready in prod by the next ship on Wednesday.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
Works fine. Only one small change in line with your own concern about `gitlabRef`

File: packages/client/components/TaskFooterIntegrateMenuSignup.tsx
Comment (nickoferrall) [reply packages/client/components/TaskFooterIntegrateMenuSignup.tsx]: I like it! I've pushed an update swapping `gitlabRef` for `integrationsRef`

**Review by Dschoordsch — APPROVED** [review]
Works for the owner of the integration. Doesn't work for non-integrated users:
* Gerrit has GitLab integration and creates a task "Push me"
* Tina is a teammate of Gerrit and tries to push the task "Push me" on behalf of Gerrit
For Jira/Git this will push the task using Gerrit's credentials and add a comment that it was pushed by Tina on his behalf. With this PR it shows a server error
```
  error: '{"data":{"createTaskIntegration":null},"errors":[{"message":"Method not implemented yet.","locations":[{"line":6,"column":3}],"path":["createTaskIntegration"]}]}'
```
I'm fine moving this into a separate issue/PR.

**Comment by nickoferrall** [comment]
Nice catch! I've implemented the `addCreatedBySomeoneElseComment` method for GitLab so that should be working now. 

I'll move this issue into the Stuck column until the base branch is merged

**Pull request merged**
